src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/LoggerAdapter.cs

```diff
@@ -41,11 +46,26 @@ public void LogError(string message, params object[] @params)
     {
 #pragma warning disable CA2254 // Template should be a static expression
         _logger.LogError(message, @params);
+
+        if (_telemetryReporter is not null)
+        {
+            using var _ = DictionaryPool<string, object>.GetPooledObject(out var props);
+
+            var index = 0;
+            foreach (var param in @params)
+            {
+                props.Add("param" + index++, param);
+            }
+
+            props.Add("message", message);
+            _telemetryReporter.ReportEvent("lsperror", VisualStudio.Telemetry.TelemetrySeverity.High, props.ToImmutableDictionary());
```

Comment:
Does `ToImmutableDictionary` use pooling? If not it seems like your pooled dict stuff above might not be buying us much?

Reply:
cc @ryzngard, that's a good point. would be good to find out what the best practices should be for this scenario.

I merged as-is since it's a nit feedback.